doc/architecture.md

```diff
@@ -0,0 +1,96 @@
+# Architecture
+
+The architecture of the system tests consist out of two layers.
+The interface between the layers are docker images.
+
+## Grammar
+
+```
+// terminals
+System         = string
+TestClass      = string
+Feature        = string
+Dockerfile     = "Dockerfile"
+TestPrefix     = "Test"
+.              = "."
+_              = "_"
+
+// Non-terminals
+SystemSpec     = System [. Feature]*
+Base           = Dockerfile . SystemSpec
+TestSpec       = TestClass | TestClass SystemSpec
+Test           = TestPrefix _ TestSpec
+```
+
+## Base Images
+
+### Naming Convention: 
+Described by Grammar: `Base`
+
+Examples:
+```
+Dockerfile.Ubuntu1804
+Dockerfile.Ubuntu1804.Boost160
+Dockerfile.Ubuntu1804.NoPETSc
+Dockerfile.Ubuntu1604
+Dockerfile.Ubuntu1604.Boost160
+```
+
+### Responsibilities:
+The image needs to provide:
```

Comment:
Use must or should here. I would say should, to make it less strict.

Reply:
My intention is to put a `must` here.

My reasoning is that by making this strict, the system tests can rely on this being installed and can thus be more generic and lightweight.
What do you think?